From 815f7705dc4f6b1c08d8369261963ceea483f297 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 6 Mar 2018 12:46:45 +0800
Subject: [PATCH 3435/5242] MLK-17156-2: ARM: dts: update dts for demo audio
 in A7 domain

commit  4033f8b63b74fe416bd687256dee477a7edc9ac8 from
https://source.codeaurora.org/external/imx/linux-imx.git

This is dts is for demo SAI + codec in A7 domain, which need to
do i2c hardware rework, connect the wm8960 to i2c7.
for this is for demo usage, so don't need the headphone plug/unplug
event, so remove the iomux for this case.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk-wm8960.dts |   23 +++--------------------
 1 file changed, 3 insertions(+), 20 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts b/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts
index 8172641..711d7b5 100644
--- a/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts
@@ -117,7 +117,7 @@
 	};
 
 	sound {
-		compatible = "fsl,imx7ulp-evk-wm8960",
+		compatible = "fsl,imx7d-evk-wm8960",
 			     "fsl,imx-audio-wm8960";
 		model = "wm8960-audio";
 		cpu-dai = <&sai0>;
@@ -125,8 +125,6 @@
 		codec-master;
 		/* JD3: hp detect high for headphone*/
 		hp-det = <3 0>;
-		hp-det-gpios = <&gpio_ptf 0 0>;
-		mic-det-gpios = <&gpio_ptf 0 0>;
 		audio-routing =
 			"Headphone Jack", "HP_L",
 			"Headphone Jack", "HP_R",
@@ -166,19 +164,6 @@
 	};
 };
 
-&iomuxc1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog_audio>;
-
-	imx7ulp-evk {
-		pinctrl_hog_audio: hoggrp-audio {
-			fsl,pins = <
-				IMX7ULP_PAD_PTF0__PTF0		0x30100
-			>;
-		};
-	};
-};
-
 &lpi2c7 {
 	status = "okay";
 
@@ -188,10 +173,8 @@
 		clocks = <&clks_m4 IMX7ULP_CLK_SCG0_CLKOUT>;
 		clock-names = "mclk";
 		wlf,shared-lrclk;
-		assigned-clocks = <&clks_m4 IMX7ULP_CM4_CLK_APLL_SEL>,
-				  <&clks_m4 IMX7ULP_CLK_SCG0_CLKOUT>;
-		assigned-clock-parents = <&clks_m4 IMX7ULP_CM4_CLK_APLL_VCO_POST_DIV2>,
-					 <&clks_m4 IMX7ULP_CM4_CLK_APLL_SEL>;
+		assigned-clocks = <&clks_m4 IMX7ULP_CLK_SCG0_CLKOUT>;
+		assigned-clock-parents = <&clks_m4 IMX7ULP_CM4_CLK_SOSC>;
 	};
 };
 
-- 
1.7.9.5

